### PR TITLE
docs(cloudflare): various updates, cleanups and fixes

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

In the cloudflare docs page I've found outdated/incorrect details which I am amending here.

Besides that I am also updating and improving some other few bits.

To see the what the various changes are please look at the commit messages.

### CODE REVIEW

File: docs/content/2.deploy/20.providers/cloudflare.md [hunk 1]
```diff
@@ -10,17 +10,15 @@ Deploy Nitro apps to CloudFlare.
 **Note:** This preset uses [service-worker syntax](https://developers.cloudflare.com/workers/learning/service-worker/) for deployment.
 ::
 
-Login to your [Cloudflare Workers](https://workers.cloudflare.com) account and obtain your `account_id` from the sidebar.
```
Comment (dario-piotrowicz) [inline docs/content/2.deploy/20.providers/cloudflare.md:13]: Developers do not need to have their account id upfront, so I feel like putting this here can just create unnecessary churn, I think it's better to just have it in the toml below where it is clear that it is an optional value

File: docs/content/2.deploy/20.providers/cloudflare.md [hunk 2]
```diff
@@ -93,41 +91,31 @@ on:
       - main
 
 jobs:
-  ci:
-    runs-on: ${{ matrix.os }}
-
-    strategy:
-      matrix:
-        os: [ ubuntu-latest ]
-        node: [ 14 ]
```
Comment (dario-piotrowicz) [inline docs/content/2.deploy/20.providers/cloudflare.md:102]: I don't think having a matrix here helped in any way?

File: docs/content/2.deploy/20.providers/cloudflare.md [hunk 3]
```diff
@@ -93,41 +91,31 @@ on:
...
-        uses: actions/checkout@master
+        uses: actions/checkout@v3
 
-      - name: Cache node_modules
-        uses: actions/cache@v2
+      - name: Setup Node.js 16.x
+        uses: actions/setup-node@v3
         with:
-          path: node_modules
-          key: ${{ matrix.os }}-node-v${{ matrix.node }}-deps-${{ hashFiles(format('{0}{1}', github.workspace, '/yarn.lock')) }}
```
Comment (dario-piotrowicz) [inline docs/content/2.deploy/20.providers/cloudflare.md:116]: I've removed the caching as it looks like an extra step unrelated to this document
(and I feel that keeping things scoped/minimal/simple makes for a better/more useful yml example here)

File: docs/content/2.deploy/20.providers/cloudflare.md [hunk 4]
```diff
@@ -10,17 +10,15 @@ Deploy Nitro apps to CloudFlare.
...
 ```ini
 name = "playground"
 main = "./.output/server/index.mjs"
 workers_dev = true
-compatibility_date = "2022-09-10"
-account_id = "<the account_id you obtained (optional)>"
-route = "<mainly useful when you want to setup custom domains (optional too)>"
+compatibility_date = "2023-12-01"
+account_id = "<(optional) your Cloudflare account id, retrievable from the Cloudflare dashboard>"
+route = "<(optional) mainly useful when you want to setup custom domains>"
```
Comment (pi0) [inline docs/content/2.deploy/20.providers/cloudflare.md:21]: Shall we comment them by default? 

```suggestion
# route = "<(optional) mainly useful when you want to setup custom domains>"
```

File: docs/content/2.deploy/20.providers/cloudflare.md [hunk 5]
```diff
@@ -10,17 +10,15 @@ Deploy Nitro apps to CloudFlare.
...
 
 ```ini
 name = "playground"
 main = "./.output/server/index.mjs"
 workers_dev = true
-compatibility_date = "2022-09-10"
-account_id = "<the account_id you obtained (optional)>"
-route = "<mainly useful when you want to setup custom domains (optional too)>"
+compatibility_date = "2023-12-01"
+account_id = "<(optional) your Cloudflare account id, retrievable from the Cloudflare dashboard>"
```
Comment (pi0) [inline docs/content/2.deploy/20.providers/cloudflare.md:20]:
```suggestion
# account_id = "<(optional) your Cloudflare account id, retrievable from the Cloudflare dashboard>"
```

File: docs/content/2.deploy/20.providers/cloudflare.md [hunk 6]
```diff
@@ -93,41 +91,31 @@ on:
...
-        with:
-          node-version: ${{ matrix.node }}
-
       - name: Checkout
-        uses: actions/checkout@master
+        uses: actions/checkout@v3
 
-      - name: Cache node_modules
-        uses: actions/cache@v2
+      - name: Setup Node.js 16.x
```
Comment (pi0) [inline docs/content/2.deploy/20.providers/cloudflare.md:101]:
```suggestion
      - name: Setup Node.js
```

File: docs/content/2.deploy/20.providers/cloudflare.md [hunk 7]
```diff
@@ -93,41 +91,31 @@ on:
...
+        uses: actions/setup-node@v3
         with:
-          path: node_modules
-          key: ${{ matrix.os }}-node-v${{ matrix.node }}-deps-${{ hashFiles(format('{0}{1}', github.workspace, '/yarn.lock')) }}
+          node-version: 16.x
```
Comment (pi0) [inline docs/content/2.deploy/20.providers/cloudflare.md:104]:
```suggestion
          node-version: 20.X
```

File: docs/content/2.deploy/20.providers/cloudflare.md (hunk 5 above)
Comment (pi0) [inline docs/content/2.deploy/20.providers/cloudflare.md:20]:
```suggestion
# account_id = "<(optional) your Cloudflare account id, retrievable from the Cloudflare dashboard>"
```

File: docs/content/2.deploy/20.providers/cloudflare.md (hunk 4 above)
Comment (dario-piotrowicz) [reply docs/content/2.deploy/20.providers/cloudflare.md]: yeah sounds good to me so that it doesn't generate errors if people just copy-paste the toml 🙂

**Comment by pi0** [comment]
Thanks for PR ❤️ (and sorry for delay on merge)

**Comment by dario-piotrowicz** [comment]
> Thanks for PR ❤️ (and sorry for delay on merge)

No problem at all, it's my pleasure, thanks for the merge 😄

**Pull request merged**